tasks/storage_deal.go

```diff
@@ -145,21 +266,37 @@ func MakeStorageDeal(ctx context.Context, config NodeConfig, node api.FullNode,
 			storagemarket.StorageDealFailing,
 			storagemarket.StorageDealError:
 
-			logStages(info, log)
+			logStages(info, de.log)
 			return errors.New("storage deal failed")
 
 		// deal is on chain, exit successfully
 		case storagemarket.StorageDealActive:
 
-			logStages(info, log)
+			logStages(info, de.log)
 			return nil
 		}
 	}
 
 	return nil
 }
 
-func logStages(info api.DealInfo, log UpdateStatus) {
+func toStageData(stage *storagemarket.DealStage) *StageData {
+	logs := make([]*Log, 0, len(stage.Logs))
+	for _, log := range stage.Logs {
+		logs = append(logs, &Log{
+			Log:       log.Log,
+			UpdatedAt: log.UpdatedTime.Time(),
```

Comment:
Better to copy the entire `Log` instead of specific fields, since this could be overlooked when adding a field to `Log`:
```go
logs := make([]*Log, len(stage.Logs))
for i, log := range stage.Logs {
    logCopy := *log
    logs[i] = &logCopy
}
```

Reply:
you can't do that because it's a different data structure

Reply:
Ah, yeah.  Oops.